bugsnag/django/__init__.py

```diff
@@ -72,36 +72,48 @@ def add_django_request_to_notification(event):
 
 
 def configure():
+    config = bugsnag.configure()
+
     # default to development if in DEBUG mode
     if getattr(settings, 'DEBUG'):
-        bugsnag.configure(release_stage='development')
-
-    request_started.connect(__track_session)
-    got_request_exception.connect(__handle_request_exception)
+        config.configure(release_stage='development')
 
     # Import Bugsnag settings from settings.py
     django_bugsnag_settings = getattr(settings, 'BUGSNAG', {})
-    bugsnag.configure(**django_bugsnag_settings)
+    config.configure(**django_bugsnag_settings)
 
-    middleware = bugsnag.configure().internal_middleware
+    middleware = config.internal_middleware
     middleware.before_notify(add_django_request_to_notification)
 
-    bugsnag.configure().runtime_versions['django'] = django.__version__
+    config.runtime_versions['django'] = django.__version__
+
+    request_started.connect(__track_session)
+    got_request_exception.connect(
+        __handle_request_exception(config.logger),
+        weak=False
+    )
```

Comment:
Django stores signal handlers as a weak reference by default, which doesn't work when this is using a closure (to have access to the logger) as it gets garbage collected. Setting `weak=False` makes it a strong reference so this works

![image](https://user-images.githubusercontent.com/282732/117943781-68894600-b304-11eb-8eab-a757c073cd8d.png)